shared/api/devlxd_storage.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+// DevLXDStoragePool is a devLXD representation of LXD storage pool.
+type DevLXDStoragePool struct {
+	// Storage pool name
+	// Example: local
+	Name string `json:"name" yaml:"name"`
+
+	// Storage pool driver
+	// Example: zfs
+	Driver string `json:"driver" yaml:"driver"`
+
+	// Pool status (Pending, Created, Errored or Unknown)
+	// Read only: true
+	// Example: Created
+	//
+	// API extension: clustering
+	Status string `json:"status" yaml:"status"`
+
+	// Storage pool configuration map (refer to doc/storage.md)
+	// Example: {"volume.block.filesystem": "ext4", "volume.size": "50GiB"}
+	Config map[string]string `json:"config" yaml:"config"`
```

Comment:
@MusicDin @markylaing we need to be very careful here as storage pool config can contain passwords to the external storage appliances. 

IIRC we only grant access to storage pool config when you are an admin, is that right @markylaing ?

@MusicDin whats the reason for needing access to this Config map?

Perhaps we need to move the pool level secret keys into the new `secrets` table and relax access to Pool config first before this change lands.